### PR TITLE
Storage Model: Interfaces and options

### PR DESCRIPTION
**Pull request by puerco**
This PR adds the simple storage backend interfaces that define the protobom storage backends.

In this change we are introducing:

1. The `storage.Backend` interface that all backends need to implement
2. The shared options struct that backends receive when Storing and Retrieving

This PR will be updated to follow the definition in the Persistence System design document:
https://docs.google.com/document/d/1Us0EMnw0bT6w24CZKHPJi1_BgQZ77zbfbSc55qymUQ0/edit

Signed-off-by: Adolfo García Veytia (Puerco) <puerco@stacklok.com>

### CODE REVIEW

File: model/v1/storage/backend.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package storage
```
Comment (jhoward-lm) [inline model/v1/storage/backend.go:1]: 😄 

```suggestion
# --------------------------------------------------------------
# SPDX-FileCopyrightText: Copyright © 2024 The Protobom Authors
# SPDX-FileType: SOURCE
# SPDX-License-Identifier: Apache-2.0
# --------------------------------------------------------------
package storage
```

File: pkg/options/options.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package options
```
Comment (jhoward-lm) [inline pkg/options/options.go:1]: 😄 

```suggestion
# --------------------------------------------------------------
# SPDX-FileCopyrightText: Copyright © 2024 The Protobom Authors
# SPDX-FileType: SOURCE
# SPDX-License-Identifier: Apache-2.0
# --------------------------------------------------------------
package options
```

**Review by jhoward-lm — APPROVED** [review]
Looks good to me!

**Comment by puerco** [comment]
OK, I've added the copyright headers. PTAL @jhoward-lm !

**Review by jhoward-lm — APPROVED** [review]

**Pull request merged**